dev/activity_diagrams/on-click-post-word-from-undo.md

```diff
@@ -0,0 +1,64 @@
+```plantuml
+
+@startuml
+
+title onClickPostWordFromUndo()
+
+participant "End User" as user
+
+box "FE" #Lightblue
+  participant "Component" as fe
+  participant "Hook" as hook
+  participant "IsLoading" as useState
+  participant "Recoil" as recoil
+box
+box "API" #Lightgreen
+  participant "API" as api
+box
+
+
+activate user
+  user -> fe: Click undo button
+  activate fe
+    fe -> hook: usePostWordFromUndo()
+    activate hook
+      hook -> useState: setLoading(true)
+      activate useState
+        useState -> useState: Sets isLoading to true
+        hook <- useState: Done
+      deactivate useState
+      hook -> recoil: getPromise(wordsFamily(undoingWordId))
+      activate recoil
+        hook <- recoil: Return word data
+      deactivate recoil
+      break if word data is undifined
```

Comment:
```suggestion
      break if word data is undefined
```

Reply:
Download this
![image](https://github.com/ajktown/docs/assets/53258958/6e99b432-de3d-4651-bb6d-0daba4e2a1a7)